rasa_core/utils.py

```diff
@@ -678,7 +681,12 @@ def from_dict(cls, data):
                 data.get("headers"),
                 data.get("basic_auth"),
                 data.get("token"),
-                data.get("token_name"))
```

Comment:
alternatively:

```
EndpointConfig(
                data.get("url"),
                data.get("params"),
                data.get("headers"),
                data.get("basic_auth"),
                data.get("token"),
                data.get("token_name"),
                **data)
```

Reply:
that makes sense. I will do the fixes this weekend 👍

Reply:
I updated the issues, however i haven't found any formatting error in Travis, not sure how to check that though

Reply:
Thanks, I will have another look on them later. Regarding travis: I think we have to let it built first. This takes usually about 20 minutes after you have pushed your changes.

Reply:
A solution to this could be:

```python
return EndpointConfig(**data)
```

and a constructor like this:

```python
class EndpointConfig(object):
    """Configuration for an external HTTP endpoint."""

    def __init__(self, url, params=None, headers=None, basic_auth=None,
                 token=None, token_name="token",**kwargs):
        self.url = url
        self.params = params if params else {}
        self.headers = headers if headers else {}
        self.basic_auth = basic_auth
        self.token = token
        self.token_name = token_name

        [setattr(self, k, v) for k, v in kwargs.items() if not hasattr(self, k)]
```

I used the `if not hasattr(self, k)` to not overwrite the dictionaries of `self.params` and `self.headers` with None.

Reply:
Regarding the formatting issues: You can also check that locally, e.g. by using a linter which knows pep8 or IDEs such as PyCharm or Visual Studio Code (with the Python plugin).